Additional CSS Should create directories

This seems to be a regression from 0.0.28, but when I put in the config that `additional-css = ["theme/reference.css"]`, it complains:

```
[havvy@ref:~/workspace/rust/reference]$ ~/.cargo/bin/mdbook build
2018-01-25 20:21:28 [INFO] (mdbook::book): Book building has started
2018-01-25 20:21:28 [INFO] (mdbook::book): Running the html backend
2018-01-25 20:21:28 [ERROR] (mdbook::utils): Error: Rendering failed
2018-01-25 20:21:28 [ERROR] (mdbook::utils):    Caused By: Unable to copy across additional CSS and JS
2018-01-25 20:21:28 [ERROR] (mdbook::utils):    Caused By: Unable to copy theme/reference.css to /home/havvy/workspace/rust/reference/book/theme/reference.css
2018-01-25 20:21:28 [ERROR] (mdbook::utils):    Caused By: No such file or directory (os error 2)
```

I would expect mdbook to actually make the `theme` directory specified and then put reference.css into it.

--- a/src/renderer/html_handlebars/hbs_renderer.rs
+++ b/src/renderer/html_handlebars/hbs_renderer.rs
@@ -245,6 +245,10 @@ impl HtmlHandlebars {
 
         for custom_file in custom_files {
             let output_location = destination.join(custom_file);
+            if let Some(parent) = output_location.parent() {
+                fs::create_dir_all(parent)
+                    .chain_err(|| format!("Unable to create {}", parent.display()))?;
+            }
             debug!(
                 "Copying {} -> {}",
                 custom_file.display(),

